connection.ts

```diff
@@ -134,12 +134,22 @@ export class Connection {
     await this.bufWriter.write(finalBuffer);
   }
 
-  async startup() {
-    const { host, port } = this.connParams;
-    this.conn = await Deno.connect({
-      port: parseInt(port, 10),
-      hostname: host,
-    });
+  public async startup() {
+    const { host, port, cert_file } = this.connParams;
+    const parsedPort = parseInt(port, 10);
+
+    if (typeof cert_file !== "undefined") {
```

Comment:
Can you add a test to what should happen if cert_file is not found locally, and if it is not valid?

Reply:
Okay. I'm assuming we want this to throw in the case that it's invalid?

Reply:
Yeah, that might be the best. Just make sure to pass a relevant throw message 👍